File: 3-leetcode/01-easy/58-length-of-last-word.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int lengthOfLastWord(string s)
    {
        if (s.length() == 1)
            return 1;

        int endIndex = s.length() - 1;

        while (endIndex > 0 && s[endIndex] == ' ')
        {
            endIndex--;
        }

        int startIndex = endIndex;

        while (startIndex > 0 && s[startIndex - 1] != ' ')
        {
            startIndex--;
        }

        return endIndex - startIndex + 1;
    }
};

File: 3-leetcode/01-easy/58-length-of-last-word.ts
const lengthOfLastWord = (s: string): number => {
  if (s.length === 1) return 1;

  let endIndex = s.length - 1;

  while (endIndex > 0 && s[endIndex] === ' ') {
    endIndex--;
  }

  let startIndex = endIndex;

  while (startIndex > 0 && s[startIndex - 1] !== ' ') {
    startIndex--;
  }

  return endIndex - startIndex + 1;
};
